April newsletter (#2071)
rcgsheffield/sheffield_hpc: newsletter/20240415.rst

**In this issue:**

- We share some of the highlights from our recent :ref:`**Research IT Forum** <research_it_forum>` event. 
- Storage Options Update: Learn about `available storage solutions <storage_options>` in light of new Google Drive policies. 
- **Expansion of the HPC Team:** Meet our newest team members, Angie and Becki who will be with us for the spring season.

.. _research_it_forum:

Research IT Forum
-----------------

Our annual Research IT Forum, held on April 9th. It provided an excellent opportunity to celebrate and showcase the groundbreaking research projects taking place on HPC. A huge thank you to all the speakers and attendees who contributed to making this year's forum a success. For those who missed it, you can find all the `presentation slides here <https://drive.google.com/drive/folders/1YPctycoYDaxZX4aqAqdCiQYQD9lgAtq_?usp=sharing>`_ . 
Also, a special thank you to Desmond Ryan and Carl Kennedy who were the key organisers this year.

.. _storage_options:

Storage Options
---------------

Research Storage
++++++++++++++++

> Recently, many of you received emails regarding Google Drive's new `15GB data cap <https://staff.sheffield.ac.uk/it-services/storage/google-data-cap-project>`_, resulting from Google's 1PB limit on university data (we currently hold 2.97PB). To address this, we encourage users to :

Author: updated

- Delete any data that is no longer needed from their google drive.

> - Migrate to our research storage solution (X-Drive), which offers up to 10TB of storage for research groups. This offers a secure, encrypted and backed up option to meet any Data storage and GDPR needs.  With access control it also facilitates collaboration and file-sharing among teams and accessible from any device that is connected to the university's VPN. Additionally, research storage can be made available on both Stanage and Bessemer upon request. It should also offer a performance boost over google drives, for users on campus or geographically close to Sheffield. Learn more about requesting or using research storage `here <https://students.sheffield.ac.uk/it-services/research/storage>`_ (You will need to be logged in, to access the page).

Reviewer: * 10TB: for free per PI; can buy more than that
* encrypted: at rest, and also in transit if not accessed over NFS
* 'to meet any Data storage and GDPR needs' -> I think we should be careful about saying things like - I suggest you discuss with Ross.

Reviewer: Simplest to trim down your 2nd bullet and rely on students.sheffield.ac.uk/it-services/research/storage to cover most points re benefits, caveats etc?

> FastData Storage

Reviewer: ```suggestion
Lustre filesystems
```

++++++++++++++++

> This month we `Fastdata <https://docs.hpc.shef.ac.uk/en/latest/hpc/filestore.html#fastdata-areas>`_ on the HPC clusters. Fast data uses a parallel, distributed file system known as lustre. Its distributed nature allows us the following benefits:

Reviewer: Hope you don't mind but I've suggested some slightly different wording for this section:

You are likely aware that the Stanage and Bessemer systems :ref:`each contain a large Lustre parallel filesystem<fastdata_dir>`.

* `/mnt/parscratch` on Stanage
* `/fastdata` on Bessemer

We would like to draw your attention to several characteristics of these Lustre filesystems

* They provide lots of temporary storage with no quotas;
* **They are not backed up;**
* They offer good performance when working with larger files than e.g. home directories - Lustre performance scales well with the number of filesystem clients, plus on Stanage connections to
 Lustre are via high-bandwidth (100 Gbps), low-latency network links;
* **They perform poorly when working with lots of small files (e.g. files only a few kB in size)** (see `this blog post <https://walkingrandomly.com/?p=6167>_`).  Even just enumerating or deleting millions of files from a Lustre filesystem can take days;
* **They perform poorly when you have a large number (>1000) of files or directories within a directory.**

Recommendations:

* Do not use these Lustre filesystems for long-term storage
* Prefer to work with larger files where possible.  For example can you combine output data within one large HDF5 file instead of creating lots of small text files?
* Try to limit the number of files or directories within a directory.
* If using interative workflows e.g. machine learning or fluid dynamics software then consider outputting data to Lustre less frequently than every time/simulation step e.g. every 1000 steps
.
* If you have concerns about how many files your workflow might create on Lustre then consider running some short tests before launching big jobs.

- Scalability: It allows us to have large amount of storage and increase it overtime without affecting the pre-existing storage.
- Allows us to achieve parallel read and writes over the multiple CPUs and nodes your jobs are running on without taking a hit to latency or throughput.
- Reliability: Allows us fault tolerance and ensures data integrity and availability.

In our case it also allows us huge amounts of storage as **currently** there are no quotas compared to the home directory.

** All this benefit come at the price of fastdata being optimised for large file operations. They are faster than Home directories and Shared (project) directories when dealing with larger files but are not performant when reading/writing lots of small files as shown on the benchmarks shown `here <https://walkingrandomly.com/?p=6167>_` **

> ** The platform team also added the command ``show-fastdata-consumption`` which shows the total usage storage by all our users on fast data. For most users this can be a proactive way of checking if its time to do some housekeeping on the fast data directories. **

Reviewer: ```suggestion
** To see how much you are currently storing on Lustre please run the new ``show-fastdata-consumption``command.  This can be a proactive way of checking if its time to do some housekeeping on some of your data hosted on Lustre. **
```

Licences
--------

> For users of licensed software such as MATLAB and Ansys, please be mindful of license availability when submitting multiple jobs that will take huge fraction of licences from the pool. Please notify us in advance if you require a significant number of licenses or consider using `job dependency <https://docs.hpc.shef.ac.uk/en/latest/hpc/scheduler/advanced_job_submission_and_control.html#dependent-jobs-on-bessemer-and-stanage>`_ to manage license usage effectively.

Author: I am going to reword it but I have added the github issue 2074 , which will address this

> For users of licensed software such as MATLAB and Ansys, please be mindful of license availability when submitting multiple jobs that will take huge fraction of licences from the pool. Please notify us in advance if you require a significant number of licenses or consider using `job dependency <https://docs.hpc.shef.ac.uk/en/latest/hpc/scheduler/advanced_job_submission_and_control.html#dependent-jobs-on-bessemer-and-stanage>`_ to manage license usage effectively.

Author: I am removing it and redo this in the future when we have found away to make the license count more transparent or whatever we come up with to solve issue 2074

New Packages and software
--------------------------

Check out the latest additions to Stanage:

- Apptainer version 1.3.0. This shouldn't result in any significant user-facing changes but please let us know if you notice any behaviour or significant performance changes, particularly relating to filesystem performance.
- Nextflow version 23.10.0

JADE2 has now been upgraded from RHEL 7.x to RHEL 8.9. Its NVIDIA libraries and tools have also been upgraded too. 

> For users using Pytorch and the GPUs to carry out experiments, using a newer Pytorch container (24.02 rather than 23.07) should fix a bug which made the H100 PCIE slower than the A100. You should see an almost 50% performance boost between the two Pytorch versions as discovered by the benchmarks Peter Haywood ran. Performance on the A100 stays the same.

Reviewer: ```suggestion
For users using Pytorch with the GPUs in Stanage, using a newer Pytorch container (24.02 rather than 23.07) should fix a bug which made the H100 GPUs  slower than the A100 GPUs. You should see an almost 50% performance boost between the two Pytorch versions as discovered by the benchmarks Peter Heywood in the `RSE team<https://rse.shef.ac.uk/>`__ ran. Performance on the A100 stays the same.
```

Author: updated

.. figure:: /images/newsletter/pytorch2402_benchmarks.png
    :width: 90%
    :align: center
    :alt: Pytorch 24.02 vs 23.07

    Pytorch 24.02 vs 23.07

.. _upcoming_training2:

Upcoming Training
-----------------

> Below are our research computing key training dates for March. You can register for these courses and more at  `Research Computing Training <https://sites.google.com/sheffield.ac.uk/research-training/>`_ . 

Author: Added the first 10 days of may.

.. _nl20240415:

News - 2024-04-15
=================

Hello everyone! We are publishing this month's newsletter a bit later than usual due to the recent festivities. We hope you all enjoyed a lovely Easter break and a happy Eid Mubarak.

**In this issue:**

- We share thoughts and materials from our recent :ref:`Research Computing @ Sheffield and Research IT Forum <recent_events_20240415>` events. 
- Storage Options Update: Learn about :ref:`available storage solutions <storage_options>` in light of new Google Drive policies. 
- **Expansion of the HPC Team:** Keep an eye out for our newest team members, Angie and Becki who will be with us for the spring season.
- :ref:`Grace Hopper GPU nodes <grace_hopper_bede>` available in the Bede HPC system.
- **Important:** Please be aware of pending OS upgrade on both Stanage and Bessemer.

Pending Operating System Upgrade on our HPC clusters
----------------------------------------------------

> The operating system on Stanage and Bessemer will be upgraded before the end of July. This upgrade ensures the security, stability and performance of our HPC infrastructure. Certain software packages and applications may not be compatible with the new OS (Rocky 9). We encourage you to start thinking about utilising newer versions of software and libraries. We will provide further updates regarding the OS upgrade as the date approaches. 

Reviewer: ```suggestion
The operating system on Stanage and Bessemer will be upgraded in the next few months. This upgrade ensures the security, stability and performance of our HPC infrastructure. Certain software packages and applications may not be compatible with the new OS (Rocky 9). We encourage you to start thinking about utilising newer versions of software and libraries. We will provide further updates regarding the OS upgrade shortly, including timelines. 
```
